src/main/java/algorithms/curated170/medium/FindPermutation.java

```diff
@@ -0,0 +1,55 @@
+package algorithms.curated170.medium;
+
+import java.util.Arrays;
+import java.util.Random;
+
+
+public class FindPermutation {
+    public static int[] findPermutation(String s) {
+        int size = s.length();
+        int[] permutation = new int[size + 1];
+
+        for (int i = 0; i <= size; ++i) {
+            permutation[i] = i + 1;
+        }
+
+        for (int i = 0; i < size; i++) {
+            if (s.charAt(i) == 'D') {
+                int left = i;
+                while (i < size && s.charAt(i) == 'D') {
+                    i++;
+                }
+                reverse(permutation, left, i);
+            }
+        }
+        return permutation;
+    }
+
+    private static void reverse(int[] nums, int start, int end) {
+        while (start < end) {
+            nums[start] ^= nums[end];
+            nums[end] ^= nums[start];
+            nums[start++] ^= nums[end--];
+            /* int temp = nums[start];
+            nums[start++] = nums[end];
+            nums[end--] = temp; */ 
+        }
```

Comment:
Klasik yöntem yerine XOR kullanma fikri muhtemelen space'ten biraz daha düşmek içindir. Hızı etkiler mi, emin değilim.
 Bu durumda pek etkisi görünmese de, daha büyük programlar için fikir vermesi açısından hoş olmuş.